Mage.Tests/src/test/java/org/mage/test/cards/cost/adventure/AdventureCardsTest.java

```diff
@@ -734,6 +734,7 @@ public void test_HostageTaker_CastFromExileAllParts() {
         checkExileCount("after exile 3", 3, PhaseStep.POSTCOMBAT_MAIN, playerA, "Curious Pair", 1);
         // play as adventure spell
         castSpell(3, PhaseStep.POSTCOMBAT_MAIN, playerA, "Treats to Share");
+        setChoice(playerA, "Hostage Taker"); // Not sure why there is an alternative there. No issue with using either. TODO: investigate?
```

Comment:
any update on this?

Reply:
So the choice the test think it has at that point are:
![image](https://github.com/magefree/mage/assets/34709007/f5edb771-9f4c-4dcf-854e-ed4e8158e499)

Treats to Share's Adventure should only be castable on its own from hand.

Testing the exact same actions than in the test, here is the first choice menu on casting the exiled by Hostage Taker adventure:
![image](https://github.com/magefree/mage/assets/34709007/1ea5f108-ea66-47a1-8de2-20cf69ea7c02)
 
Clicking the first choice, second menu to choose approving object (like the test):
![image](https://github.com/magefree/mage/assets/34709007/96569b75-4395-42a8-ad54-689d80214e4f)
Both option lead to the adventure side starting being cast, with any color being accepted for the {G} payment.

Clicking the second choice, no menu to choose approving object, and start of cast of the creature side (with any type modification).

So really I'm not sure what is incorrect there for there to be two Approving objects. That's the only weird situation I've noticed among many manual tests (+ unit tests already existing or created in the PR)

Reply:
Oh actually that might just be Hostage Taker needing its own MageIdentifier. Will try to do that.

Reply:
Oh that goes deep in the "you may play|you may cast" with modifiers. If that's alright with you, I'll keep that investigation for the #10573 rework making use of the current AsThough changes to differentiate the modifications on how cards may be cast.

Reply:
Yes, that's fine to leave it as is for this PR and track separately for future fix

Reply:
made a note on that PR's big task list to remember.